src/frontend/src/flows/authenticate/index.ts

```diff
@@ -175,26 +189,61 @@ export interface AuthSuccess {
  * Internet Identity window.
  */
 export default async (): Promise<AuthSuccess> => {
-  return new Promise((resolve) => {
-    withLoader(async () => {
-      const authContext = await waitForAuthRequest();
-      if (authContext === null) {
-        // The user has manually navigated to "/#authorize".
-        window.location.hash = "";
-        window.location.reload();
-        return;
-      }
-      const userNumber = getUserNumber();
-      init(authContext, userNumber).then(resolve);
+  const authContext = await withLoader(() => waitForAuthRequest());
+  if (authContext === null) {
+    // The user has manually navigated to "/#authorize".
+    window.location.hash = "";
+    window.location.reload();
+    return new Promise((_resolve) => {
+      // never resolve, window is being reloaded
     });
+  }
+
+  const validationResult = await withLoader(
+    async () =>
+      await validateDerivationOrigin(
+        authContext.requestOrigin,
+        authContext.authRequest.derivationOrigin
+      )
+  );
+  if (validationResult.result === "invalid") {
```

Comment:
Would be nice to also check for `valid` explicitly and/or make sure the match is exhaustive in case more alternatives are added, see this for instance: https://github.com/dfinity/internet-identity/blob/1939b306a307e62a2b0f2e76ea74cea983a0e6d4/src/frontend/src/flows/recovery/recoverWith/device.ts#L54-L65

Reply:
Ah, right. Will do 👍